docs/developer/highlevel_design.rst

```diff
@@ -0,0 +1,220 @@
+
+* The Front End (UI)
+
+    * Stages - Stages group and control the user interaction
+    * Widgets - There is a widget for each filter/function to handle the user interaction with the specific filter/function.
+
+* The Backend
+
+    * iMars3D Filter/Function Library
+    * iMars3D Interactive Workflow Engine
+
+
+iMars3D Auto Reconstruction Application
+=======================================
+
+iMars3D auto reconstruction application and the application to be used when performing auto reconstruction.
+
+.. note:: Auto reconstruction is equivalent to auto reduction. It is invoked in the same way automatically after data acquisition.
+
+Auto reconstruction takes an already specified configuration json file which contains all the information required to run a reconstruction process.
+
+Structure
+---------
+
+The iMars3D auto reconstruction application si comprised of:
+
+* The Backend
+
+    * iMars3D Filter/Function Library
+    * IMars3D non-interactive Workflow Engine
+
+This application could also be run from the command line.
+
+For auto reconstruction a wrapper script is required.
+The system that invokes the auto reconstruction assumes an interface that takes the location of the raw ct_scans.
+
+iMars3D Library
+===============
+
+The iMars3D (filter/function) library is simply the set of filters and functions involved in performing a reconstruction process.
+The library is expected to be used within a python application.
+It is the application that keeps track of the state of the reconstruction and maintains the intermediate results as required to transition from one filter/function to the next.
+
+iMars3D Key Design Decisions
+############################
+
+There is a UI widget for each filter/function.
+The widget handles the interaction with the user to collect the data (parameters) required to execute the filter/function, if any.
+
+During Interactive reconstruction the UI software will wait for the user to execute the filter before saving the associated configuration (parameter) data.
+
+The option to save to disk the result of running a filter/function (intermediate data) will be a checkbox.
+The intermediate data will be saved when the user selects to save the configuration data.
+If the box is checked prior to running the filter it is ignored.
+
+The widget associated with the filter/function calls the interactive reconstruction engine method ``run_single()`` and serializes itself a dictionary entry.
+
+There will be two buffers to hold results.
+The current buffer holds the accepted result from the previous filter/function.
+A temporary buffer holds the result of the current filter/function execution.
+This is to allow the user the ability to reject the intermediate result if desired.
+If/when the intermediate result from executing the current filter/function is satisfactory it will be copied to the current buffer and the temporary buffer will be used for the next filter/function to execute.
```

Comment:
this section is perfect for the enabler that implements the logic in the main window application.